### PR TITLE
docs: add how to build the docs

### PR DESCRIPTION
**Pull request by david-caro**
* Add also the requires.txt to the docs directory, with the
  dependencies to build the docs themselves

Signed-off-by: David Caro <david.caro@cern.ch>

### CODE REVIEW

File: docs/building_the_docs.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
+..
+    This file is part of INSPIRE.
+    Copyright (C) 2015, 2016 CERN.
```
Comment (jmartinm) [inline docs/building_the_docs.rst]: We just put the year(s) the file existed, so in this case just 2016.

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+six
```
Comment (jmartinm) [inline docs/requirements.txt]: You can add the docs requirements directly in the `docs` section of `setup.py` - https://github.com/inspirehep/inspire-next/blob/master/setup.py#L94

Then you can mention in the docs that they can be installed with `pip install -e .[docs]`

**Comment by jmartinm** [comment]
To start getting used to the Invenio style of commit log messages, @kaplun can you put the pointer to the latest documentation on commit log etiquette? We should probably link to it from our documentation too.

**Comment by kaplun** [comment]
It's here: http://invenio.readthedocs.io/en/latest/technology/git.html#r2-remarks-on-commit-log-messages

File: docs/requirements.txt (hunk 2 above)
Comment (david-caro) [inline docs/requirements.txt]: I'll play with it, but the first try installed all the inspire-next deps too (flask and such)

**Comment by jacquerie** [comment]
LGTM for content; minor nits for the commit message (space after colon, no capital letter after colon) and the year in the license.

**Comment by jacquerie** [comment]
FYI: if you `git push --force` N times, N builds are added to the queue on Travis, potentially clogging it, and N messages are added by Coveralls, which results in `N*len(developers)` emails :D

**Pull request merged**